src/rustinfo.rs

```diff
@@ -85,29 +85,42 @@ fn load_setup(rust_info: &mut RustInfo) {
                     }
                 }
 
+                let mut triple = String::new();
+
                 let mut value = values.remove("target_arch");
-                if value.is_some() {
-                    rust_info.target_arch = Some(value.unwrap());
+                if let Some(value) = value {
+                    triple.push_str(&value);
+                    rust_info.target_arch = Some(value);
                 }
 
-                value = values.remove("target_env");
-                if value.is_some() {
-                    rust_info.target_env = Some(value.unwrap());
+                value = values.remove("target_vendor");
+                if let Some(value) = value {
+                    triple.push_str("-");
```

Comment:
if target_arch is empty it means triple will now start as '-xxxxxx'.
maybe worth to have a standalone function that sets that logic from a rust info struct into that same struct.
that way you can validate and do it correctly and populate it the right way (with default as none of course).
and we can have a really good unit test for that function.

Reply:
I was half way through to implement a standalone function when I realized setting `target_arch` to unknown, would fix this.